evennia/contrib/plural.py

```diff
@@ -0,0 +1,190 @@
+"""
+Light contrib to group objects by name and display plural names.
+
+Evennia contrib - Vincent Le Goff 2017
+
+This contrib allows to group object names (in 'look' for instance)
+using their singualr names.  You could have a display varying from:
+    You see: a rock, a rock
+To:
+    You see: a rock X 2
+You could even customize it to have:
+    You see: 2 rocks
+
+To install, you simply need to import and inherit from the `PluralNames`
+mixin in your typeclasses.  Be sure to put it first in your inheritance
+tree.  You could do it for all your typeclasses, or just your rooms for
+instance.
+
+```python
+# Example with 'typeclasses/rooms.py'
+from evennia import DefaultRoom
+
+from evennia.contrib.plural import PluralNames
+
+class Room(PluralNames, DefaultRoom):
+
+    # ...
+```
+
+The `PluralNames` is a very light mixin that will override your
+`return_apperance` hook.  You can, however, write your own `return_appearance`
+hook and use the `group_objects` directly (see below for more information).
+
+To change the way your objects will display plural names, you can
+override the `get_plural_name` method (see the default implementation
+below).  You can do so with all your typeclasses, if you don't like the default.
+
+"""
+
+def group_objects(objects, attrname="key"):
+    """
+    Return the grouped list of objects, using their singular names.
+
+    Args:
+        objects (list ob Object): list of objects to be grouped.
+        attrname (str, optional): the name of the singular attribute.
+
+    Returns:
+        grouped (list): the same list of objects grouped by names.
+        The list will contain tuples: the number (int),
+        the object (Object) and the list of all objects of the same name.
+
+    Notes:
+        In the case of grouping (if an entry has more than 1 object),
+        only one of these objects will be displayed for grouping.
+        This is usually the expected behavior.
+
+        Grouping will occur based on the `attrname` argument.  By
+        default, the objects' `key` are used as singular names.  You
+        can provide a different attribute names.  It could be linked
+        to a property, or an Evennia attribute (just give it
+        "db.<name of attribute>" as value).
```

Comment:
Actually, best is to reverse this syntax since it will then match up with the way prototypes are defined. That is, start your name with `db_` to refer to a property, (like `db_key`), otherwise it is considered an Attribute.